osgar/tools/calibfish.py

```diff
@@ -0,0 +1,84 @@
+import cv2
+import glob
+from pprint import pprint
+import numpy as np
+
+PATTERN = (8,5)
+
+def capture(camera=0):
+    cap = cv2.VideoCapture(camera)
+    if not cap.isOpened():
+        return
+
+    print("backend:", cap.getBackendName())
+    cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+    cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
+    cap.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"MJPG"))
+
+    count = 0
+    while True:
+        status, img = cap.read()
+        if status == False:
+            continue
+        gray = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+        gray = cv2.resize(gray, (img.shape[1]//2, img.shape[0]//2))
+        found, corners = cv2.findChessboardCorners(gray, PATTERN, cv2.CALIB_CB_ADAPTIVE_THRESH+cv2.CALIB_CB_FAST_CHECK+cv2.CALIB_CB_NORMALIZE_IMAGE)
+        img_copy = img.copy()
+        if found:
+            corners *= 2
+            cv2.drawChessboardCorners(img, PATTERN, corners, found)
+        cv2.imshow('img', img)
+        key = cv2.waitKey(1)
+        if key == 27:
+            return
+        if key == ord('s'):
+            if found:
+                filename = 'capture{:02d}.jpg'.format(count)
+                print(filename)
+                cv2.imwrite('images/'+filename, img_copy)
+                count += 1
+
+
+def calibrate(images):
+    subpix_criteria = cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 30, 0.1
+    calibration_flags = cv2.fisheye.CALIB_RECOMPUTE_EXTRINSIC + cv2.fisheye.CALIB_CHECK_COND + cv2.fisheye.CALIB_FIX_SKEW
+    # prepare object points, like (0,0,0), (1,0,0), (2,0,0) ....,(PATTERN[0],PATTERN[1],0)
+    # but for some crazy reasons it needs to have the shape (1, x, 3)
+    objp = np.zeros((1, PATTERN[0]*PATTERN[1],3), np.float32)
+    objp[0,:,:2] = np.mgrid[0:PATTERN[0], 0:PATTERN[1]].T.reshape(-1,2)
+    objpoints = [] # 3d point in real world space
+    imgpoints = [] # 2d points in image plane.
+    for image in sorted(images):
+        gray = cv2.imread(image, cv2.IMREAD_GRAYSCALE)
+        found, corners = cv2.findChessboardCorners(gray, PATTERN, cv2.CALIB_CB_ADAPTIVE_THRESH+cv2.CALIB_CB_FAST_CHECK+cv2.CALIB_CB_NORMALIZE_IMAGE)
+        print(image, found)
+        if found:
+            objpoints.append(objp)
+            cv2.cornerSubPix(gray, corners, (3,3), (-1,-1), subpix_criteria)
+            imgpoints.append(corners)
+    K = np.zeros((3, 3))
+    D = np.zeros((4, 1))
+    rms, _, _, _, _ = cv2.fisheye.calibrate(objpoints, imgpoints, gray.shape[::-1], K, D,
+        flags=calibration_flags, criteria=(cv2.TERM_CRITERIA_EPS+cv2.TERM_CRITERIA_MAX_ITER, 30, 1e-6))
+
+    return len(objpoints), rms, gray.shape[::-1], K, D
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Calibrate fisheye camera')
+    parser.add_argument('--camera', help='camera id', type=int, default=0)
+    args = parser.parse_args()
+
+    images = glob.glob('images/capture*.jpg')
```

Comment:
this is bit strange that it behaves based on some status folder and in that case args.camera is ignored

Reply:
It works with current folder, so yes. The `args.camera` is used only when there are no images pre-captured. Do you have some other behavior in mind that you would like more?